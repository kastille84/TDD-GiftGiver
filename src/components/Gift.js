import React, {Component} from 'react'
import {
  Form, FormGroup, FormControl, ControlLabel, Button
} from 'react-bootstrap';

class Gift extends Component {

  state={
    person: '',
    present:''
  }
  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <Form.Label>Person</Form.Label>
            <FormControl 
              className="input-person"
              onChange={event=> this.setState({person: event.target.value})}
            />
          </FormGroup>
          <FormGroup>
            <Form.Label>Present</Form.Label>
            <FormControl 
              className="input-present"
              onChange={event=> this.setState({present: event.target.value})}
            />
          </FormGroup>
        </Form>
        <Button 
          className="btn-remove"
          onClick={() => this.props.removeGift(this.props.gift.id)}
        >
          Remove Gift
        </Button>
      </div>
    )
  }
}

export default Gift;