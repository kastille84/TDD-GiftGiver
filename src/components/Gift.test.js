import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Gift from './Gift';

configure({adapter: new Adapter()});

const id = 1;
const mockRemove = jest.fn();
//used to mimick the props given to the real Gift comp by App comp
const props = {
  gift: {id: id}, 
  removeGift: mockRemove
}
const gift = shallow(<Gift {...props} />);

describe("Gift", () => {

  it('should render properly', () => {
    expect(gift).toMatchSnapshot();
  });
  it('initializes a perosn and present in `state', () => {
    expect(gift.state()).toEqual({person: '', present: ''})
  })
})

describe('when typing into the person input', () => {
  const person = 'Uncle';
  beforeEach(() => {
    gift.find('.input-person').simulate('change', {target: {value: person}})
  })
  it('updates the person in `state`', () => {
    expect(gift.state().person).toEqual(person)
  })
})

describe('when typing into the present intput', () => {
  const present = "xbox";
  beforeEach(() => {
    gift.find('.input-present').simulate('change', {target:{value: present}})
  })
  it(`updates the present in state`, () => {
    expect(gift.state().present).toEqual(present);
  })
})

describe('when clicking the "Remove Gift" button', () => {
  beforeEach(() => {
    gift.find('.btn-remove').simulate('click')
  })

  it("calls the removeGift callback", () => {
    //jest's Mock Function, defined  at top 
    expect(mockRemove).toHaveBeenCalledWith(id);
  })
}) 