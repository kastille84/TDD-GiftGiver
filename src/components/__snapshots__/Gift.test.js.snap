// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift should render properly 1`] = `ShallowWrapper {}`;
